backend/dataall/modules/dataset_sharing/services/share_item_service.py

```diff
@@ -63,7 +63,11 @@ def revoke_items_share_object(uri, revoked_uris):
                 resource_uri=dataset.datasetUri,
             )
 
-            ShareNotificationService.notify_share_object_rejection(session, context.username, context.email_id, dataset, share)
+            ShareNotificationService(
+                session=session,
+                dataset=dataset,
+                share=share
+            ).notify_share_object_rejection(email_id=context.username)
```

Comment:
Do we need the email_id=context.username field if we are using the share.owner which is also the email id of the user creating the share ?

Reply:
I think we do need to pass `context.username` to know who is performing the operation - i.e. for approvals/rejections/revokes what user rejected or approved the share? (this could be any member of the dataset SamlAdminGroup or stewards - or even requestorTeam revoking some items)

We use that information for the email message sent to the respective users/groups

Reply:
Yeah that's right!. I get it now. The message in the email body/ UI notification needs to have that information

Reply:
It is passed as email_id, because I wanted to account for your suggestion @TejasRGitHub an make it clear that the username is actually the email. Just in case I checked the definition of the "notify-X" functions and they use the email_id for to create the message: `def notify_share_object_submission(self, email_id: str):`